File: tests/java/Service.java
package com.example;

import java.util.List;
import java.util.Map;

public class UserService {
    private Map<String, User> users;

    public UserService() {
        this.users = new Map<>();
    }

    public User createUser(String name) {
        return new User(name);
    }

    public User getUser(String id) {
        return users.get(id);
    }

    public void deleteUser(String id) {
        users.remove(id);
    }

    public List<User> listUsers() {
        return List.of();
    }
}

class AdminService extends UserService {
    public void createAdmin(String name) {
        // admin creation
    }
}

File: tests/kotlin/Service.kt
package com.example

import java.util.HashMap

class UserService {
    private val users = HashMap<String, User>()

    fun createUser(name: String): User {
        val id = users.size + 1
        val user = User(id, name)
        users[id.toString()] = user
        return user
    }

    fun getUser(id: String): User? {
        return users[id]
    }

    fun deleteUser(id: String) {
        users.remove(id)
    }
}

class AdminService : UserService() {
    fun createAdmin(name: String, role: String): User {
        return createUser(name)
    }
}

data class User(val id: Int, val name: String)
